tests/cluster-compose.yml

```diff
@@ -0,0 +1,199 @@
+      - --cluster-announce-port 7001
+      - --cluster-announce-bus-port 17001
+      - --protected-mode no
+    ports:
+      - "7001:7001"
+      - "17001:17001"
+    volumes:
+      - ./data/7001:/data
+
+  redis-node-2:
+    image: "redis:8.0"
+    command:
+      - redis-server
+      - --port 7002
+      - --cluster-enabled yes
+      - --cluster-config-file nodes.conf
+      - --cluster-node-timeout 5000
+      - --appendonly yes
+      - --cluster-announce-ip host.docker.internal
+      - --cluster-announce-port 7002
+      - --cluster-announce-bus-port 17002
+      - --protected-mode no
+    ports:
+      - "7002:7002"
+      - "17002:17002"
+    volumes:
+      - ./data/7002:/data
+
+  redis-node-3:
+    image: "redis:8.0"
+    command:
+      - redis-server
+      - --port 7003
+      - --cluster-enabled yes
+      - --cluster-config-file nodes.conf
+      - --cluster-node-timeout 5000
+      - --appendonly yes
+      - --cluster-announce-ip host.docker.internal
+      - --cluster-announce-port 7003
+      - --cluster-announce-bus-port 17003
+      - --protected-mode no
+    ports:
+      - "7003:7003"
+      - "17003:17003"
+    volumes:
+      - ./data/7003:/data
+
+  redis-node-4:
+    image: "redis:8.0"
+    command:
+      - redis-server
+      - --port 7004
+      - --cluster-enabled yes
+      - --cluster-config-file nodes.conf
+      - --cluster-node-timeout 5000
+      - --appendonly yes
+      - --cluster-announce-ip host.docker.internal
+      - --cluster-announce-port 7004
+      - --cluster-announce-bus-port 17004
+      - --protected-mode no
+    ports:
+      - "7004:7004"
+      - "17004:17004"
+    volumes:
+      - ./data/7004:/data
+
+  redis-node-5:
+    image: "redis:8.0"
+    command:
+      - redis-server
+      - --port 7005
+      - --cluster-enabled yes
+      - --cluster-config-file nodes.conf
+      - --cluster-node-timeout 5000
+      - --appendonly yes
+      - --cluster-announce-ip host.docker.internal
+      - --cluster-announce-port 7005
+      - --cluster-announce-bus-port 17005
+      - --protected-mode no
+    ports:
+      - "7005:7005"
+      - "17005:17005"
+    volumes:
+      - ./data/7005:/data
+
+  redis-node-6:
+    image: "redis:8.0"
+    command:
+      - redis-server
+      - --port 7006
+      - --cluster-enabled yes
+      - --cluster-config-file nodes.conf
+      - --cluster-node-timeout 5000
+      - --appendonly yes
+      - --cluster-announce-ip host.docker.internal
+      - --cluster-announce-port 7006
+      - --cluster-announce-bus-port 17006
+      - --protected-mode no
+    ports:
+      - "7006:7006"
+      - "17006:17006"
+    volumes:
+      - ./data/7006:/data
+
+  redis-cluster-setup:
+    image: "redis:8.0"
+    depends_on:
+      - redis-node-1
+      - redis-node-2
+      - redis-node-3
+      - redis-node-4
+      - redis-node-5
+      - redis-node-6
+    entrypoint:
+      - sh
+      - -c
+      - |
+        echo "Waiting for Redis nodes to be available..."
```

Comment:
This is a little much, but it was the result of many iterations on cluster startup failures. I may simplify!